ain/ain.py

```diff
@@ -1,38 +1,219 @@
+        self,
+        blockHashOrBlockNumber: Union[str, int],
+    ) -> str:
+        """
+        A coroutine returns the address of the forger of given block
+        """
+        if type(blockHashOrBlockNumber) is str:
+            return await self.provider.send(
+                "ain_getProposerByHash", {"hash": blockHashOrBlockNumber}
+            )
+        elif type(blockHashOrBlockNumber) is int:
+            return await self.provider.send(
+                "ain_getProposerByNumber", {"number": blockHashOrBlockNumber}
+            )
+        else:
+            raise TypeError("blockHashOrBlockNumber has invalid type")
+
+    async def getValidators(
+        self,
+        blockHashOrBlockNumber: Union[str, int],
+    ):
+        """
+        A coroutine returns the list of validators for a given block
+        """
+        if type(blockHashOrBlockNumber) is str:
+            return await self.provider.send(
+                "ain_getValidatorsByHash", {"hash": blockHashOrBlockNumber}
+            )
+        elif type(blockHashOrBlockNumber) is int:
+            return await self.provider.send(
+                "ain_getValidatorsByNumber", {"number": blockHashOrBlockNumber}
+            )
+        else:
+            raise TypeError("blockHashOrBlockNumber has invalid type")
 
-        return txBody
+    async def getTransaction(self, transactionHash: str):
+        """
+        Returns the transaction with the given transaction hash.
+        """
+        return await self.provider.send(
+            "ain_getTransactionByHash", {"hash": transactionHash}
+        )
 
-    async def sendTransaction(self, transactionObject):
+    async def getStateUsage(self, appName: str):
+        """
+        Returns the transaction with the given transaction hash.
+        """
+        return await self.provider.send(
+            "ain_getStateUsage", {"app_name": appName}
+        )
+
+    async def sendTransaction(self, transactionObject: TransactionInput):
+        """
+        Signs and sends a transaction to the network
+        """
         txBody = await self.buildTransactionBody(transactionObject)
-        signature = ecSignTransaction(txBody, self.wallet.defaultAccount.privateKey)
-        result = await self.provider.send("ain_sendSignedTransaction", {"signature": signature, "tx_body": txBody})
-        return result
+        signature = self.wallet.signTransaction(
+            txBody, getattr(transactionObject, "address", None)
+        )
+        return await self.sendSignedTransaction(signature, txBody)
+
+    async def sendSignedTransaction(self, signature: str, txBody: TransactionBody):
+        """
+        Sends a signed transaction to the network
+        """
+        return await self.provider.send(
+            "ain_sendSignedTransaction", {"signature": signature, "tx_body": txBody}
+        )
+
+    async def sendTransactionBatch(self):
+        """
+        Sends signed transactions to the network.
+        """
+        pass
```

Comment:
could you leave TODOs for the functions that haven't been implemented?

Reply:
Added. PTAL